Sources/Network/JSON.swift

```diff
@@ -186,6 +186,61 @@ public enum JSON {
         return value
     }
 
+    /// Parse an attribute of type `T: Mappable` with the given key on the given JSON dictionary, validating if
+    /// the value can be used to instantiate a Mappable model. Additionally, an optional `parseAPIError` closure can be supplied
+    /// so that if a parsing step fails an attempt is made to extract a domain specific error and throw it.
+    ///
+    /// - Parameters:
+    ///   - type: The type of the attribute to parse.
+    ///   - key: The JSON attribute key to parse.
+    ///   - json: The JSON dictionary.
+    ///   - parseAPIError: The API error parsing closure.
+    /// - Returns: The value of type `T` associated to the given attribute's key.
+    /// - Throws: An error of type `JSON.Error`, or a domain specific `Swift.Error` produced by `parseAPIError`.
+    public static func parseMappableAttribute<T: Mappable>(
+        _ type: T.Type,
+        key: JSON.AttributeKey,
+        json: JSON.Dictionary,
+        parseAPIError: ParseAPIErrorClosure? = nil) throws -> T {
```

Comment:
could we also add the `where predicate: ParsePredicateClosure<T>? = nil` parameter?
I think it can come useful in some scenarios 😉 
(same applies for other calls below)